doc/content/xapi/internals/certificates.md

```diff
@@ -0,0 +1,105 @@
+```
+
+
+## Fundamental Certificates
+
+Certificates that identify a host. These certificates are comprised of
+both a private and a public key. The public key may be distributed to
+other hosts.
+
+### xapi-ssl.pem
+
+This certificate identifies a host for extra-pool clients.
+
+This is the certificate used by the API HTTPS server that clients like
+XenCenter or CVAD connect to. On installation of XenServer it is auto
+generated but can be updated by a user using the API. This is the most
+important certificate for a user to establish an HTTPS connection to a
+pool or host to be used as an API.
+
+* /etc/xensource/xapi-ssl.pem
+* contains private and public key for this host
+* `Host.get_server_certificate` API call
+* referenced by /etc/stunnel/xapi.conf
+* `xe host-server-certificate-install` XE command to replace the
+  certificate.
+* See below for xapi-stunnel-ca-bundle for additional certificates that
+  can be added to a pool in support of a user-supplied host certificate.
+
+### `xapi-pool-tls.pem`
+
+This certificate identifies a host inside a pool. It is auto generated
+and used for all intra-pool HTTPS connections. It needs to be
+distributed inside a pool to establish trust. The distribution of the
+public part of the certificate is performed by the API and must not be
+done manually.
+
+* /etc/xensource/xapi-pool-tls.pem
+* contains private and public key for this host
+* referenced by /etc/stunnel/xapi.conf
+* This certificate can be re-generated using the API or XE
+* `Host.refresh_server_certificate`
+* `xe host-refresh-server-certificate`
+
+## Certificate Bundles
+
+Certifiacte bundles are used by stunnel. They are a collection of public
+keys from hosts and certificates provided by a user. Knowing a host's
+public key facilitates stunnel connecting to the host.
+
+Certificates in a pool are
+
+### `xapi-pool-ca-bundle.pem` and `certs-pool/*.pem`
+
+Collection of public keys from xapi-pool-tls.pem across the
+pool. The public keys are collected in the certs-pool directory: each is
+named after the UUID of its host and the bundle is constructed from
+them.
+
+* bundle of public keys from hosts' `xapi-pool-tls.pem`
+* constructed from PEM files in `certs-pool/`
+* `/opt/xensource/bin/update-ca-bundle.sh` generates bundle from PEM files
+* used by
+
+### `xapi-stunnel-ca-bundle.pem` and `certs/*.pem`
+
+User-installed certificates; they are not essential for the operation of
+a pool from Xapi's perspective. They make stunnel aware of certificates
+used by clients when using HTTPS for API calls.
+
+* in a plain pool installation, these are empty
+* bundle of public keys provided by a user
+* constructed from PEM files in `certs/`
+* `/opt/xensource/bin/update-ca-bundle.sh` generates bundle from PEM files
+* Updated by a user using `xe pool-install-ca-certificate`
+* `Pool.install_ca_certificate`
+* `Pool.uninstall_ca_certificate`
+* `xe pool-certificate-sync` explicitly distribute these certificates in
+  the pool.
```

Comment:
Xapi is able to use client certificates as well, although not in use anymore. They were used for the cloud project